console - ReferenceError: ejs:1 ("Sorry, something bad happened")

Hi there,

my last issue was closed, but the error still appears...

I installed Etherpad on CentOS according to this tutorial: https://www.digitalocean.com/community/tutorials/how-to-install-etherpad-for-production-with-node-js-and-mysql-on-a-vps


I did a fresh install - unfortunately I'm getting this error:

[2015-01-04 18:08:20.111] [ERROR] console - ReferenceError: ejs:1
 >> 1| <% e._init(buf); %><%
    2|   var settings = require("ep_etherpad-lite/node/utils/Settings");
    3| %>
    4| <!doctype html>

buf is not defined
    at eval (eval at <anonymous> (/home/etherpad/etherpad-lite/src/node_modules/ejs/lib/ejs.js:281:12), <anonymous>:2:145)
    at /home/etherpad/etherpad-lite/src/node_modules/ejs/lib/ejs.js:307:17
    at Object.exports.render (/home/etherpad/etherpad-lite/src/node_modules/ejs/lib/ejs.js:166:13)
    at Object.exports.require (/home/etherpad/etherpad-lite/src/node/eejs/index.js:122:17)
    at /home/etherpad/etherpad-lite/src/node/hooks/express/specialpages.js:15:19
    at callbacks (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:164:37)
    at param (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:138:11)
    at pass (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:145:5)
    at Router._dispatch (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:173:5)
    at Object.router (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:33:10)


If I start etherpad, I'm getting this error:

[etherpad@pad etherpad-lite]$ ./bin/run.sh
Ensure that all dependencies are up to date...  If this is the first time you have run Etherpad please be patient.
npm WARN package.json async-stacktrace@0.0.2 No repository field.
npm WARN package.json channels@0.0.4 No repository field.
npm WARN package.json tinycon@0.0.1 No repository field.
Ensure jQuery is downloaded and up to date...
Clear minfified cache...
Ensure custom css/js files are created...
Started Etherpad...
[2015-01-04 18:11:08.173] [INFO] console - Installed plugins: ep_etherpad-lite
[2015-01-04 18:11:08.192] [INFO] console - Your Etherpad git version is e2ea82f
[2015-01-04 18:11:08.192] [INFO] console - Report bugs at https://github.com/ether/etherpad-lite/issues
[2015-01-04 18:11:08.320] [INFO] console - You can access your Etherpad instance at http://0.0.0.0:9001/
[2015-01-04 18:11:08.321] [WARN] console - Admin username and password not set in settings.json.  To access admin please uncomment and edit 'users' in settings.json



If I try to open etherpad via browser, the browser-output is:

{
  "error": "Sorry, something bad happened!"
}


The console shows:

[2015-01-04 18:11:50.853] [ERROR] console - ReferenceError: ejs:1
 >> 1| <% e._init(buf); %><%
    2|   var settings = require("ep_etherpad-lite/node/utils/Settings");
    3| %>
    4| <!doctype html>

buf is not defined
    at eval (eval at <anonymous> (/home/etherpad/etherpad-lite/src/node_modules/ejs/lib/ejs.js:281:12), <anonymous>:2:8000)
    at /home/etherpad/etherpad-lite/src/node_modules/ejs/lib/ejs.js:307:17
    at Object.exports.render (/home/etherpad/etherpad-lite/src/node_modules/ejs/lib/ejs.js:166:13)
    at Object.exports.require (/home/etherpad/etherpad-lite/src/node/eejs/index.js:122:17)
    at /home/etherpad/etherpad-lite/src/node/hooks/express/specialpages.js:15:19
    at callbacks (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:164:37)
    at param (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:138:11)
    at pass (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:145:5)
    at Router._dispatch (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:173:5)
    at Object.router (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:33:10)
[2015-01-04 18:11:50.920] [INFO] ueberDB - Flushed 1 values


This error was on Debian 7 and now on CentOS - the VPS was always a fresh setup.

Use the official docs please.  If the problem still persists while using these docs let us know
----- Reply message -----
From: "SPQRInc" <notifications@github.com>
To: "ether/etherpad-lite" <etherpad-lite@noreply.github.com>
Subject: [etherpad-lite] console - ReferenceError: ejs:1 ("Sorry, something bad happened") (#2437)
Date: Sun, Jan 4, 2015 15:13


Hi there,

my last issue was closed, but the error still appears...

I installed Etherpad on CentOS according to this tutorial: https://www.digitalocean.com/community/tutorials/how-to-install-etherpad-for-production-with-node-js-and-mysql-on-a-vps

I did a fresh install - unfortunately I'm getting this error:

[2015-01-04 18:08:20.111] [ERROR] console - ReferenceError: ejs:1

1| <% e._init(buf); %>Test
2| <% e._exit(); %>

buf is not defined
at eval (eval at (/home/etherpad/etherpad-lite/src/node_modules/ejs/lib/ejs.js:281:12), :2:145)
at /home/etherpad/etherpad-lite/src/node_modules/ejs/lib/ejs.js:307:17
at Object.exports.render (/home/etherpad/etherpad-lite/src/node_modules/ejs/lib/ejs.js:166:13)
at Object.exports.require (/home/etherpad/etherpad-lite/src/node/eejs/index.js:122:17)
at /home/etherpad/etherpad-lite/src/node/hooks/express/specialpages.js:15:19
at callbacks (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:164:37)
at param (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:138:11)
at pass (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:145:5)
at Router._dispatch (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:173:5)
at Object.router (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:33:10)

If I start etherpad, I'm getting this error:

[etherpad@pad etherpad-lite]$ ./bin/run.sh
Ensure that all dependencies are up to date... If this is the first time you have run Etherpad please be patient.
npm WARN package.json async-stacktrace@0.0.2 No repository field.
npm WARN package.json channels@0.0.4 No repository field.
npm WARN package.json tinycon@0.0.1 No repository field.
Ensure jQuery is downloaded and up to date...
Clear minfified cache...
Ensure custom css/js files are created...
Started Etherpad...
[2015-01-04 18:11:08.173] [INFO] console - Installed plugins: ep_etherpad-lite
[2015-01-04 18:11:08.192] [INFO] console - Your Etherpad git version is e2ea82f<https://github.com/ether/etherpad-lite/commit/e2ea82f8df23f7ba68b5873024ddef178748c8c9>
[2015-01-04 18:11:08.192] [INFO] console - Report bugs at https://github.com/ether/etherpad-lite/issues
[2015-01-04 18:11:08.320] [INFO] console - You can access your Etherpad instance at http://0.0.0.0:9001/
[2015-01-04 18:11:08.321] [WARN] console - Admin username and password not set in settings.json. To access admin please uncomment and edit 'users' in settings.json

If I try to open etherpad via browser, the browser-output is:

{
"error": "Sorry, something bad happened!"
}

The console shows:

[2015-01-04 18:11:50.853] [ERROR] console - ReferenceError: ejs:1

1| <% e._init(buf); %><%
2| var settings = require("ep_etherpad-lite/node/utils/Settings");
3| %>
4| <!doctype html>

buf is not defined
at eval (eval at (/home/etherpad/etherpad-lite/src/node_modules/ejs/lib/ejs.js:281:12), :2:8000)
at /home/etherpad/etherpad-lite/src/node_modules/ejs/lib/ejs.js:307:17
at Object.exports.render (/home/etherpad/etherpad-lite/src/node_modules/ejs/lib/ejs.js:166:13)
at Object.exports.require (/home/etherpad/etherpad-lite/src/node/eejs/index.js:122:17)
at /home/etherpad/etherpad-lite/src/node/hooks/express/specialpages.js:15:19
at callbacks (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:164:37)
at param (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:138:11)
at pass (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:145:5)
at Router._dispatch (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:173:5)
at Object.router (/home/etherpad/etherpad-lite/src/node_modules/express/lib/router/index.js:33:10)
[2015-01-04 18:11:50.920] [INFO] ueberDB - Flushed 1 values

This error was on Debian 7 and now on CentOS - the VPS was always a fresh setup.

—
Reply to this email directly or view it on GitHub<https://github.com/ether/etherpad-lite/issues/2437>.

Sry, but same issue here:
```bash
$ node --version
v0.10.35
$ git describe
1.5.0-1-gcbfc833

$ ./bin/run.sh 
Ensure that all dependencies are up to date...  If this is the first time you have run Etherpad please be patient.
npm WARN package.json async-stacktrace@0.0.2 No repository field.
npm WARN package.json channels@0.0.4 No repository field.
npm WARN package.json tinycon@0.0.1 No repository field.
Ensure jQuery is downloaded and up to date...
Clear minfified cache...
Ensure custom css/js files are created...
Started Etherpad...
[2015-01-04 20:09:40.482] [WARN] console - DirtyDB is used. This is fine for testing but not recommended for production.
[2015-01-04 20:09:42.147] [INFO] console - Installed plugins: ep_etherpad-lite
[2015-01-04 20:09:42.182] [INFO] console - Your Etherpad git version is cbfc833
[2015-01-04 20:09:42.183] [INFO] console - Report bugs at https://github.com/ether/etherpad-lite/issues
[2015-01-04 20:09:42.375] [INFO] console - You can access your Etherpad instance at http://0.0.0.0:9001/
[2015-01-04 20:09:42.376] [INFO] console - The plugin admin page is at http://0.0.0.0:9001/admin/plugins
[2015-01-04 20:09:55.051] [ERROR] console - ReferenceError: ejs:1
 >> 1| <% e._init(buf); %><%
    2|   var settings = require("ep_etherpad-lite/node/utils/Settings");
    3| %>
    4| <!doctype html>

buf is not defined
    at eval (eval at <anonymous> (/home/etherpad/local/src/etherpad-lite/src/node_modules/ejs/lib/ejs.js:281:12), <anonymous>:2:8000)
    at /home/etherpad/local/src/etherpad-lite/src/node_modules/ejs/lib/ejs.js:307:17
    at Object.exports.render (/home/etherpad/local/src/etherpad-lite/src/node_modules/ejs/lib/ejs.js:166:13)
    at Object.exports.require (/home/etherpad/local/src/etherpad-lite/src/node/eejs/index.js:122:17)
    at /home/etherpad/local/src/etherpad-lite/src/node/hooks/express/specialpages.js:15:19
    at callbacks (/home/etherpad/local/src/etherpad-lite/src/node_modules/express/lib/router/index.js:164:37)
    at param (/home/etherpad/local/src/etherpad-lite/src/node_modules/express/lib/router/index.js:138:11)
    at pass (/home/etherpad/local/src/etherpad-lite/src/node_modules/express/lib/router/index.js:145:5)
    at Router._dispatch (/home/etherpad/local/src/etherpad-lite/src/node_modules/express/lib/router/index.js:173:5)
    at Object.router (/home/etherpad/local/src/etherpad-lite/src/node_modules/express/lib/router/index.js:33:10)

```

Okay what the fsck...

Was developing plugins, restarted maybe 50 times and then this suddenly came up..  No updates or anything..

```
[2015-01-04 20:49:55.595] [ERROR] console - ReferenceError: ejs:1
 >> 1| <% e._init(buf); %><%
    2|   var settings = require("ep_etherpad-lite/node/utils/Settings")
    3|     , hooks = require('ep_etherpad-lite/static/js/pluginfw/hooks')
    4|     , langs = require("ep_etherpad-lite/node/hooks/i18n").availableLangs

buf is not defined
    at eval (eval at <anonymous> (/home/jose/etherpad-lite/src/node_modules/ejs/lib/ejs.js:281:12), <anonymous>:2:21291)
    at /home/jose/etherpad-lite/src/node_modules/ejs/lib/ejs.js:307:17
    at Object.exports.render (/home/jose/etherpad-lite/src/node_modules/ejs/lib/ejs.js:166:13)
    at Object.exports.require (/home/jose/etherpad-lite/src/node/eejs/index.js:122:17)
    at /home/jose/etherpad-lite/src/node/hooks/express/specialpages.js:40:19
    at callbacks (/home/jose/etherpad-lite/src/node_modules/express/lib/router/index.js:164:37)
    at param (/home/jose/etherpad-lite/src/node_modules/express/lib/router/index.js:138:11)
    at paramCallback (/home/jose/etherpad-lite/src/node_modules/express/lib/router/index.js:150:30)
    at /home/jose/etherpad-lite/src/node/hooks/express/padurlsanitize.js:27:11
    at /home/jose/etherpad-lite/src/node/db/PadManager.js:217:9
```
No updates etc..  I will investigate..

--- a/src/node/eejs/index.js
+++ b/src/node/eejs/index.js
@@ -114,7 +114,7 @@ exports.require = function (name, args, mod) {
 
   args.e = exports;
   args.require = require;
-  var template = '<% e._init(buf); %>' + fs.readFileSync(ejspath).toString() + '<% e._exit(); %>';
+  var template = '<% e._init([__output]); %>' + fs.readFileSync(ejspath).toString() + '<% e._exit(); %>';
   
   exports.info.args.push(args);
   exports.info.file_stack.push({path: ejspath, inherit: []});

--- a/src/package.json
+++ b/src/package.json
@@ -28,7 +28,7 @@
                       "cheerio"                 : ">=0.18.0",
                       "async-stacktrace"        : ">=0.0.2",
                       "npm"                     : ">=2.1.x",
-                      "ejs"                     : "~1.0.0",
+                      "ejs"                     : "^2.0.0",
                       "graceful-fs"             : ">=3.0.4",
                       "slide"                   : ">=1.1.6",
                       "semver"                  : ">=2.3.0",

